src/arch/xtensa/include/arch/lib/cache.h

```diff
@@ -22,7 +22,26 @@
 
 #define SRAM_UNCACHED_ALIAS	0x20000000
 
+#ifdef CONFIG_IMX
+/*
+ * Every 512M in 4GB space has dedicate cache attribute.
+ * 1: write through
+ * 2: cache bypass
+ * 4: write back
+ * F: invalid access
+ */
+extern uint32_t _memmap_cacheattr_reset;
+
+#define _addr_range(address) (((uintptr_t)(address) >> 29) & 0xF)
```

Comment:
AND with 0x7? Only 3 bits are of interest to us, especially since you're shifting 29 bits out (the topmost bit is always zero).

Reply:
a comment with natural language description of what we are doing here is welcomed.

Reply:
Fwiw, the linker script is pre-processed so you could set the ```memmap_cacheattr_reset``` using the same macros for shift sizes/bits that sre used to construct the ```is_cacheable()``` macro.